epistemix_platform/tests/services/test_s3_results_repository.py

```
@@ -0,0 +1,550 @@
+    - "s3://bucket/results/job_12/run_4.zip"
+  When I extract the key from each URL
+  Then I get "results/job_12/run_4.zip" for all formats
+
+Scenario 7: Reject invalid S3 URL formats
+  Given an invalid S3 URL "https://example.com/file.zip"
+  When I call get_download_url
+  Then a ValueError is raised
+  And the error message indicates invalid URL format
+"""
+
+from unittest.mock import MagicMock, patch
```

Flake8 (7.3.0)

[error] 'unittest.mock.patch' imported but unused

(F401)